src/llvm/lib_llvm/initialization.cr

```diff
@@ -0,0 +1,18 @@
+require "./types"
+
+lib LibLLVM
+  {% if LibLLVM::IS_LT_170 %}
```

Comment:
**polish:** IIUC this header file is completely gone in LLVM >=17.0, so I figure `{% skip_file unless LibLLVM::IS_LT_170 %}` might be more appropriate than branching the entire lib body?

Ditto for `pass_builder.cr` and `pass_manager_builder.cr`.